### PR TITLE
Add functionality to show and hide an events channel

### PR DESCRIPTION
**Pull request by social-anthrax**
# New Functionality
**New Mod command**: Mods will now be able to show and hide an events channel to simplify running weekly events so that the server doesn't get cluttered. 

# Added
## [Utils.hs](../blob/autoEvent/src/Utils.hs#L369)
- Added function to convert any snowflake into an integer
- Added function to move Channel to a different position in guild given guild, channel, and location
- Added handler for mod only commands
## [ModifyEventsChannel.hs](../blob/autoEvent/src/receivers/ModifyEventsChannel.hs)
- Added receiver for showEvents
- Added function that handles showing and hiding events
  - Moves the position of a category
  - Changes visibility settings for @everyone 
# Changes
## [Utils.hs](../blob/autoEvent/src/Utils.hs#L245)
- Changed the way Moderators are verified to accept a list of role names.
- Added error function to reduce code duplication

### CODE REVIEW

File: src/receivers/ModifyEventsChannel.hs [hunk 1]
```diff
@@ -10,33 +10,37 @@ import Control.Monad ( void )
...
-    -- void $ restCall $ EditChannelPermissions 837451182946517054 0x000000400 undefined
+    sendMessageChan (messageChannel m) $ owoify "Moving Events Channel."
+
+    mbchannelObj <- restCall $ GetChannel eventsChannelId
+    case mbchannelObj of 
+        Left err -> liftIO $ print err
+        Right mbchannelObj -> do
+            case mbchannelObj of
+                ChannelGuildCategory _ guild _ position _ -> do
+                    if position == 0
```
Comment (yutotakano) [inline src/receivers/ModifyEventsChannel.hs]: As this is quite repetitive and very long, could we rewrite it to something like the following? It's still over 100 characters due to indents, but much more legible.

```hs
let selector = if position == 0 then fst else snd
let permOpts = ChannelPermissionOpts { channelPermissionOptsAllow = selector (0, 0x000000400)
                                     , channelPermissionOptDeny   = selector (0x000000400, 0)
                                     , channelPermissionsOptsType = ChannelPermissionsOptsRole
                                     }
restCall $ R.EditChannelPermissions eventChannelId guild permOpts
moveChannel guild eventsChannelId $ selector (99, 0)
```

File: src/receivers/ModifyEventsChannel.hs [hunk 2]
```diff
@@ -10,33 +10,37 @@ import Control.Monad ( void )
...
-        then do
-            moveChannel (fromJust (messageGuild m)) 99 -- Arbitrarily large number to shift to bottom
-        else do
-            moveChannel (fromJust (messageGuild m)) 0
-
-
-    -- void $ restCall $ EditChannelPermissions 837451182946517054 0x000000400 undefined
+    sendMessageChan (messageChannel m) $ owoify "Moving Events Channel."
+
+    mbchannelObj <- restCall $ GetChannel eventsChannelId
```
Comment (yutotakano) [inline src/receivers/ModifyEventsChannel.hs]: Both `GetChannel` and `EditChannelPermissions` are re-exported from `Discord.Requests`, which is available under the `R` alias (i.e. `R.GetChannel`, and `R.EditChannelPermissions`). No need to import the internal `Discord.Internal.Rest.Channel` module.

File: src/receivers/ModifyEventsChannel.hs [hunk 3]
```diff
@@ -10,33 +10,37 @@ import Control.Monad ( void )
...
+    case mbchannelObj of 
+        Left err -> liftIO $ print err
+        Right mbchannelObj -> do
```
Comment (yutotakano) [inline src/receivers/ModifyEventsChannel.hs]: I don't understand the rationale behind the variable naming choice here? It's not a Maybe (it's an Either), plus it's reusing the same name twice?

**Review by yutotakano — CHANGES_REQUESTED** [review]
Looked over it, the long lines can do a bit of change. Kinda trying to veer away from explicit spaghetti code you know? 
The PR text is nicely formatted and easy to understand, great! The links definitely help.

File: src/receivers/ModifyEventsChannel.hs (hunk 3 above)
Comment (social-anthrax) [reply src/receivers/ModifyEventsChannel.hs]: The naming choice here is leftover from when I was using Maybe, I'll fix that now.

File: src/receivers/ModifyEventsChannel.hs (hunk 1 above)
Comment (social-anthrax) [reply src/receivers/ModifyEventsChannel.hs]: This is much neater, thank you for the suggestion!

Comment (jacobjwalters) [reply src/receivers/ModifyEventsChannel.hs]: Could the braced parts not be moved onto a new line to avoid massive indents?

File: src/Utils.hs [hunk 4]
```diff
@@ -215,15 +218,8 @@ sendFileChan :: ChannelId -> T.Text -> FilePath -> DiscordHandler ()
 sendFileChan c name fp = do
     mFileContent <- liftIO $ safeReadFile fp
     case mFileContent of
-        Nothing          -> do
-<<<<<<< HEAD
-            _ <- liftIO $ putStrLn $ "[WARN] Couldn't load file: " <> fp
```
Comment (jacobjwalters) [inline src/Utils.hs:220]: Please can we keep this error message, I added it yesterday cos we keep on having problems with asset loading and it's nice to see what we're trying to read

File: src/receivers/ModifyEventsChannel.hs (hunk 1 above)
Comment (social-anthrax) [reply src/receivers/ModifyEventsChannel.hs]: Yes, I'll shift it onto the next line to make it more readable

**Review by yutotakano — APPROVED** [review]
LGTM!

File: src/Utils.hs (hunk 4 above)
Comment (social-anthrax) [reply src/Utils.hs:220]: Seems to have been accidentally removed during a merge, fixed now.

**Comment by social-anthrax** [comment]
Will leave branch up for now, as I would like to try and make the events channel an editable config file so that any guild can declare an events category.

**Comment by social-anthrax** [comment]
Looking into the possibility of getting previous roles and then using bitwise operations to modify the role rather hard setting it as is happening now.

**Pull request merged**